katana/utils/user_utils.py

```diff
@@ -3,6 +3,82 @@
 from django.conf import settings
 from utils.navigator_util import Navigator
 
+from .directory_traversal_utils import join_path, file_or_dir_exists
+
+DOTDATA = '.data'
+WAPPLOGS = 'wapp_logs'
+WAPPSDATA = 'wapps_data'
+USERDATA = 'user_data'
+
+
+class UserData:
+    # constructor for class User_data
+    def __init__(self, wapp_name, username=None):
+        self.username = username
+        self.wapp_name = wapp_name
+
+    def get_user_path(self, request):
+        username = user_authenticated(request)
+        if username is not None:
+            self.username = username
+            top_dir = get_user_home_dir(username)
+            if top_dir is not "":
+                user_dir = join_path(top_dir, USERDATA, username)
+                return user_dir
+            else:
+                print('fetching of top level directory resulted in a empty string')
+                return None
+        else:
+            print('user not authenticated')
+            return None
+
+    # define method, get .data directory
+    # given request object
+    def get_dotdata_dir(self, request):
+        # get location to .data
+        if self.wapp_name is not None:
+            wapp_name = self.wapp_name
+            # get top level directory for user_data
+            user_path = self.get_user_path(self, request)
+            if user_path is not None:
+                dotdata_dir = join_path(user_path, WAPPSDATA, wapp_name, DOTDATA)
+                if file_or_dir_exists(dotdata_dir):
```

Comment:
I think instead of just verifying if the directory exists, your function should also create the directory if it does not exist.